### ISSUE TITLE
[Bug]: Only the className attribute is supported

### ISSUE THREAD

**Issue by LoganDark** [tailwindlabs/headlessui#696]
### What package within Headless UI are you using?

@headlessui/react

### What version of that package are you using?

1.3.0

### What browser are you using?

Chrome

### Reproduction repository

N/A

### Describe your issue

Most (all?) of your components only accept className as a prop. Any other attributes, like `style`, or [`react-scoped-css` `data-v-...` attributes](https://github.com/gaoxiaoliangz/react-scoped-css/), are completely ignored.

I use a pattern in my own codebase for passing all unrecognized props through - here is an example:

```tsx
function LinkButton({onClick, children, ...others}: {onClick: React.MouseEventHandler<HTMLAnchorElement>, [key: string]: any}, ref: any) {
	const callback = useCallback(e => {
		e.preventDefault()
		return onClick(e)
	}, [onClick])

	// eslint-disable-next-line jsx-a11y/anchor-is-valid
	return <a href="#" ref={ref} onClick={callback} {...others}>{children}</a>
}
```

(Yes, I know. As ESLint says, using links as buttons is bad. It's just an example. tailwindcss/typography does not style buttons, or support making buttons look like links.)

So basically, you use a destructuring assignment to extract all the props you explicitly handle, include `[key: string]: any` in your prop type, and include `...others` in your destructuring expression so that you can then use `{...others}` to apply them back to your element.

Let me know if you would accept a PR that fixes this.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

All the props are already forwarded, passing through `data-...` attributes or the `style` attribute should just work. Here is an example: https://codesandbox.io/s/lucid-ellis-0r97x?file=/src/App.js
If this is not the case for you, please open a new issue with a minimal reproduction repo.

**Comment by mericmo** [comment]
![image](https://user-images.githubusercontent.com/8073550/199179166-5c49ddc9-2fac-4fb9-9dbc-f127b8f10c65.png)
The style in my headless component is not effective. Other pages in my project are OK. I haven't solved this problem for two days. Please

**Comment by yshrsmz** [comment]
Same here, 
I'm using @headlessui/vue v1.7.4 with vue 3.2.41
